localstack-core/localstack/services/sqs/provider.py

```diff
@@ -1232,8 +1233,11 @@ def receive_message(
 
         num = max_number_of_messages or 1
 
-        # backdoor to get all messages
-        if num == -1:
+        # override receieve count with value from custom header
+        if override := extract_message_count_from_headers(context):
+            num = override
+        elif num == -1:
```

Comment:
... and revert this for the above change to work
```suggestion
        # backdoor to get all messages
        if num == -1:
```